### DISCUSSION TITLE
Unable to enable sanitizer on macOS

### DISCUSSION THREAD

**Discussion by dpo** [mesonbuild/meson#11905] [General]
Hi. I'm building a mixed C-C++-Fortran project with Meson and trying to enable the address sanitizer. I use the flag `Db_sanitize=address,undefined`. I've also tried `-Db_sanitize=address -Db_lundef=false`. Invariably, the build results in
```
ld: library not found for -lasan
```
Indeed, I cannot find a `libasan` on my system. I use gcc/g++/gfortran version 13 installed with Homebrew. I tried the arm64 and intel versions. Both fail with the same error message.

I have tried using the clang/clang++/gfortran-13 combination with the clang that ships with macOS, but sadly, clang does not support OpenMP, which is needed in the project.

I guess I'm a bit confused as to how one is supposed to be using the address sanitizer with gcc on macOS exactly. Could someone provide any guidance here?

Thank you in advance.

**Comment by eli-schwartz** [comment]
This is usually supposed to be handled internally by the compiler with the `-fsanitize=` argument. Build systems like meson don't pass `-lasan` directly. So I have a sneaking suspicion that this means the compiler toolchain itself lacks sufficient support for asan.

**Reply by eli-schwartz** [reply]
Okay, looking around this seems to be https://github.com/orgs/Homebrew/discussions/3384

Maybe specifically unsupported by the arm64 edition of homebrew gcc? Does that make sense?

**Comment by dpo** [comment]
Thank you. I get the same error with the compilers for the Intel architecture, though.

**Reply by dpo** [reply]
And the same happens with the gcc-12 toolchain.
